Created three new generators

ajchili opened the pull request.

ajchili committed:
Created VersionNumberGenerator

ajchili committed:
VersionNumberGenerator -> StringVersionNumberGenerator

ajchili committed:
Created StringCarGenerator

ajchili committed:
Created StringIPAddressGenerator

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringVersionGenerator.java:
```diff
@@ -0,0 +1,78 @@
+import tk.zielony.randomdata.DataContext;
+import tk.zielony.randomdata.Generator;
+
+/**
```
remove automatic comment

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringVersionGenerator.java:
```diff
@@ -0,0 +1,78 @@
+                + (random.nextInt(maxBuild - minBuild) + minBuild);
+    }
+
+    public int nextVersionNumber(@Nullable DataContext context) {
```
this is pretty much an int generator. I'm not sure if it's neccessary

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringVersionGenerator.java:
```diff
@@ -0,0 +1,78 @@
+        return random.nextInt(maxVersion - minVersion) + minVersion;
+    }
+
+    public int nextBuildNumber(@Nullable DataContext context) {
```
same here

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringVersionGenerator.java:
```diff
@@ -0,0 +1,78 @@
+ * Created by kirinpatel on 12/1/17.
+ */
+
+public class StringVersionGenerator extends Generator<String> {
```
This generator could be more generic with options to specify a custom separator and any number of sections.

ZieIony commented on util/src/test/java/tk/zielony/randomdata/util/StringVersionGeneratorText.java:
```diff
@@ -0,0 +1,25 @@
+
+import static org.junit.Assert.assertEquals;
+
+/**
```
remove automatic comment

ZieIony commented on util/src/test/java/tk/zielony/randomdata/util/StringVersionGeneratorText.java:
```diff
@@ -0,0 +1,25 @@
+    @Test
+    public void generationIsCorrect() throws Exception {
+        StringVersionGenerator versionNumberGenerator = new StringVersionGenerator();
+        int version = versionNumberGenerator.nextVersionNumber(null);
```
avoid passing nulls as parameters. If this parameter is not used, remove it. If it's not used in some cases, provide an overloaded method with no parameter.

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringCarGenerator.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String nextInternal(@Nullable DataContext context) {
+        int numOfCarTypes = Car.class.getEnumConstants().length;
```
there's no need to use reflection - each enum has an array of its items. Use: `Car.values().length`

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringCarGenerator.java:
```diff
@@ -0,0 +1,23 @@
+    @Override
+    public String nextInternal(@Nullable DataContext context) {
+        int numOfCarTypes = Car.class.getEnumConstants().length;
+        Car c = Car.class.getEnumConstants()[random.nextInt(numOfCarTypes)];
```
same here: `Car.values[random.nextInt(numOfCarTypes)]`

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringCarGenerator.java:
```diff
@@ -0,0 +1,23 @@
+ * Created by kirinpatel on 12/1/17.
+ */
+
+public class StringCarGenerator extends Generator<String> {
```
This generator could be replaced with a generic enum generator. Cars are a broad topic. Type of car body could come with model, manufacturer names, engines, etc. in a form of a data base. All of these attributes should be linked using DataContext.

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/Car.java:
```diff
@@ -0,0 +1,9 @@
+ */
+
+public enum Car {
+    Sedan, Hatchback, Crossover, Van, Minivan, Truck, SUV, Exotic, Convertible
```
It's more like a car body type, not a car.

ZieIony reviewed (changes requested):
In general, this pull request should be split into 3 separate pull requests. IP addresses are fine. Version numbers could be more generic. Cars won't make it in the current form - it's a broad topic and it could easily be a package with all types of generators allowing one to get random data for an app for car sellers or racing fans.

ajchili committed:
Removed two of three generators and refined the third

StringCarGenerator and StringVersionGenerator were removed.
StringIPAddressGenerator had classRange switched to enum and a test added.
Automatic comments were also remvoed.

ZieIony commented on util/src/test/java/tk/zielony/randomdata/util/StringIPAddressGeneratorTest.java:
```diff
@@ -0,0 +1,31 @@
+            int initialSection = Integer.parseInt(ipAddress.substring(0, ipAddress.indexOf('.')));
+            switch(range) {
+                case A:
+                    assertEquals(true, 191 >= initialSection && initialSection >= 128);
```
these checks could be simplified using enum methods as suggested in one of the previous comments

ZieIony commented on util/src/test/java/tk/zielony/randomdata/util/StringIPAddressGeneratorTest.java:
```diff
@@ -0,0 +1,31 @@
+        StringIPAddressGenerator ipAddressGenerator = new StringIPAddressGenerator();
+        for (StringIPAddressGenerator.classRange range : StringIPAddressGenerator.classRange.values()) {
+            ipAddressGenerator.setClassRange(range);
+            String ipAddress = ipAddressGenerator.nextInternal(null);
```
tests should use Generator.next() instead of Generator.nextInternal(). I should probably make Generator.nextInternal() protected so it's only used by subclasses what was my initial intention

ZieIony reviewed (changes requested):
Thanks for the update and forgive me the delay. I've just added some comments. I believe that would make this PR ready for approval.

ajchili committed:
Implemented suggested changes

ajchili commented:
I also saw that I used the incorrect ranges for IP Address Classes so that has changed as well.

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringIPAddressGenerator.java:
```diff
@@ -9,38 +9,42 @@
-    public enum classRange {
-        A, B, C
+    public enum ClassRange {
+        A(1, 126), B(128, 191), C(192, 223); // check the values, I'm not sure if these are okay
```
this comment was for you to check :D It shouldn't stay in the code

ZieIony commented on util/src/main/java/tk/zielony/randomdata/util/StringIPAddressGenerator.java:
```diff
@@ -9,38 +9,42 @@
+    public enum ClassRange {
+        A(1, 126), B(128, 191), C(192, 223); // check the values, I'm not sure if these are okay
+
+        int min,max;
```
formatting

ajchili replied:
How would you like this to be formatted? With new lines?

ZieIony commented on util/src/test/java/tk/zielony/randomdata/util/StringIPAddressGeneratorTest.java:
```diff
@@ -11,21 +9,11 @@
-                    assertEquals(true, 127 >= initialSection && initialSection >= 1);
-                    break;
-            }
+            assertEquals(true, range.max >= initialSection && initialSection >= range.min);
```
min and max should be read-only. Please make them private, add accessors and use them
